### PR TITLE
[13.2.X] Introduce PCL-like workflows for SiStrip Lorentz Angle Monitoring (and potentially calibration)

### PR DESCRIPTION
**Pull request by mmusich**
backport of https://github.com/cms-sw/cmssw/pull/42574
backport of https://github.com/cms-sw/cmssw/pull/42680

From the original PRs:

#### PR description:

This PR introduces a PCL-like workflow for SiStrip Lorentz Angle Monitoring (and potentially calibration), by using the `SiStripCalCosmics` dataset in input. The setup closesly resemble the one presented multiple times (see e.g. [this talk](https://indico.cern.ch/event/1271467/contributions/5350054/attachments/2624636/4538685/20230404_RWalsh_StripsCalibLA.pdf)) in which the hall mobility of the strips modules (only TIB and TOB for the time being, grouped in regions sharing same layer / module type) is estimated by fitting the distribution of the Strip cluster multiplicity as a function of the track grazing angle.  
The  `SiStripCalCosmics`  AlCaReco producer is added to the `ExpressCosmics` PD in `autoAlCa` while the new `PromptCalibProdSiStripLA` ALCAPrompt producer is introduced in several configuration files in  `Configuration/StandardSequences`  package. 
As a minimal follow-up, in order to be able to support uploads from the Prompt Calibration Loop:
   * updated `DropBoxMetaData` writer and reader configuration files
   * provided steering meta-data for upload to prod / prep and for the multi-run harvesting case
   * added `SiStripLorentzAngle` workflow to `Calibration/TkAlCaRecoProducers/test/testPCLAlCaHarvesting.py`
   
#### PR validation:
   
Run the following command:

```
cmsDriver.py testReAlCa -s ALCA:PromptCalibProdSiStripLA --conditions 130X_dataRun3_Prompt_v2 --scenario cosmics --data --era Run3_2023 --datatier ALCARECO --eventcontent ALCARECO --processName=ReAlCa -n 100000 --dasquery='file dataset=/Cosmics/Commissioning2023-SiStripCalCosmics-PromptReco-v1/ALCARECO run=364141' --nThreads=4 
```
followed by:

```
cmsDriver.py stepHarvest -s ALCAHARVEST:SiStripLA --conditions 130X_dataRun3_Prompt_v2 --scenario pp --data --era Run3_2023 -n -1 --filein file:PromptCalibProdSiStripLA.root
```

and obtained reasonable output DQM and db files.
Also run successfully the augmented relval wf 1005.0 (`runTheMatrix.py -l 1005.0 -t 4 -j 8 --ibeos`.  
   
`scram b runtests` runs fine.   

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Combined backport of https://github.com/cms-sw/cmssw/pull/42574 and https://github.com/cms-sw/cmssw/pull/42680
While this PR is not strictly needed for the rest of the 2023 data-taking ppRef / HI runs, it would be nice to have it included in the next 13.2.X (if that's possible) in order to profit of the testing machinery setup at Tier-0, in order to ask for a dedicated replay.

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflow = 1005.0

**Comment by mmusich** [comment]
please test

File: CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLHarvester.cc [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+#include "DQMServices/Core/interface/DQMEDHarvester.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
```
Comment (perrotta) [inline CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLHarvester.cc:39]: duplicate
 ```suggestion
```

File: CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+#include "DataFormats/TrackerRecHit2D/interface/SiStripMatchedRecHit2D.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiStripRecHit1D.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiStripRecHit2D.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/StripGeomDetUnit.h"
+#include "Geometry/TrackerGeometryBuilder/interface/StripGeomDetUnit.h"
```
Comment (perrotta) [inline CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc:47]: Duplicate
```suggestion
```

File: CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+  // first collect all the clusters
+  for (const auto& assoc : *trajTrackAssociations) {
+    const auto traj = assoc.key.get();
+    const auto track = assoc.val.get();
+
+    iHists_.h1_["track_pt"]->Fill(track->pt());
+    iHists_.h1_["track_eta"]->Fill(track->eta());
+    iHists_.h1_["track_phi"]->Fill(track->phi());
+    iHists_.h1_["track_validhits"]->Fill(track->numberOfValidHits());
+    iHists_.h1_["track_chi2ndof"]->Fill((track->chi2() / track->ndof()));
```
Comment (perrotta) [inline CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc:251]: You could define some
```
    const auto normChi2 = track->chi2() / track->ndof();
```
or even 
```
    const auto normChi2 = track->ndof() > 0 ? track->chi2() / track->ndof() : -1.;
```
and use consistently to fill the histos

**Review by perrotta — COMMENTED** [review]
While reviewing, a few minor (and fully innocuous) typos were spot.
They are NOT intended to be implemented in this backport, but can be taken into account for a possible follow up PR in the master

**Comment by perrotta** [comment]
+alca
- Verbatim backport of https://github.com/cms-sw/cmssw/pull/42574 and https://github.com/cms-sw/cmssw/pull/42680
- Only the dedicated ExpressCosmics PD is touched (wf 1005 among the tested ones), as planned
- I think this is worth being backported for the next data taking 13_2_X release

**Comment by perrotta** [comment]
+pd

**Comment by srimanob** [comment]
+upgrade

**Comment by sunilUIET** [comment]
+pdmv

**Comment by mmusich** [comment]
>  They are NOT intended to be implemented in this backport, but can be taken into account for a possible follow up PR in the master

thanks @perrotta I'll take care of these later on in master.

**Comment by mmusich** [comment]
@cms-sw/operations-l2 this backport is missing your signature

**Comment by perrotta** [comment]
urgent

**Comment by mmusich** [comment]
@antoniovilela @rappoccio this PR is fully signed (waiting your `operations` signature) and despite being flagged `urgent`, is not yet merged. Can you clarify the foreseen integration path ?

**Comment by antoniovilela** [comment]
+1

File: CalibTracker/SiStripLorentzAngle/interface/SiStripLorentzAngleCalibrationHelpers.h [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      subdet = "TOB";
+      layer = tTopo->layer(mod);
+    }
+
+    std::string type = (detid.stereo() ? "s" : "a");
+    std::string d_l_t = Form("%s_L%d%s", subdet.c_str(), layer, type.c_str());
+
+    if (layer == 0)
+      return subdet;
+    return d_l_t;
```
[the remark this answers is not in the document]
Comment (mmusich) [reply CalibTracker/SiStripLorentzAngle/interface/SiStripLorentzAngleCalibrationHelpers.h:38]: Addressed at https://github.com/mmusich/cmssw/commit/19bf615f8649660683e0652f5289fb7f3e5d0d93

File: CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc (hunk 2 above)
Comment (mmusich) [reply CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc:47]: Addressed at https://github.com/mmusich/cmssw/commit/19bf615f8649660683e0652f5289fb7f3e5d0d93

File: CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc (hunk 3 above)
Comment (mmusich) [reply CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLMonitor.cc:251]: Addressed at https://github.com/mmusich/cmssw/commit/19bf615f8649660683e0652f5289fb7f3e5d0d93

File: CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLHarvester.cc (hunk 1 above)
Comment (mmusich) [reply CalibTracker/SiStripLorentzAngle/plugins/SiStripLorentzAnglePCLHarvester.cc:39]: Addressed at https://github.com/mmusich/cmssw/commit/19bf615f8649660683e0652f5289fb7f3e5d0d93

**Pull request merged**